src/Protocol/Diaspora.php

```diff
@@ -1600,28 +1600,55 @@ public static function isRedmatrix($url)
 	private static function plink($addr, $guid, $parent_guid = '')
 	{
 		$contact = Contact::getDetailsByAddr($addr);
+		if (empty($contact)) {
+			Logger::info('No contact data for address', ['addr' => $addr]);
+			return '';
+		}
 
-		// Fallback
-		if (!$contact) {
-			if ($parent_guid != '') {
-				return "https://" . substr($addr, strpos($addr, "@") + 1) . "/posts/" . $parent_guid . "#" . $guid;
-			} else {
-				return "https://" . substr($addr, strpos($addr, "@") + 1) . "/posts/" . $guid;
+		if (empty($contact['baseurl'])) {
+			$contact['baseurl'] = 'https://' . substr($addr, strpos($addr, '@') + 1);
+			Logger::info('Create baseurl from address', ['baseurl' => $contact['baseurl'], 'url' => $contact['url']]);
+		}
+
+		$platform = '';
+		$gserver = DBA::selectFirst('gserver', ['platform'], ['nurl' => Strings::normaliseLink($contact['baseurl'])]);
+		if (!empty($gserver['platform'])) {
+			$platform = strtolower($gserver['platform']);
+			Logger::info('Detected platform', ['platform' => $platform, 'url' => $contact['url']]);
+		}
+
+		if (!in_array($platform, ['diaspora', 'friendica', 'hubzilla', 'socialhome'])) {
+			if (self::isHubzilla($contact['url'])) {
+				Logger::info('Detected unknown platform as Hubzilla', ['platform' => $platform, 'url' => $contact['url']]);
+				$platform = 'hubzilla';
+			} elseif ($contact['network'] == Protocol::DFRN) {
+				Logger::info('Detected unknown platform as Friendica', ['platform' => $platform, 'url' => $contact['url']]);
+				$platform = 'friendica';
 			}
 		}
 
-		if ($contact["network"] == Protocol::DFRN) {
-			return str_replace("/profile/" . $contact["nick"] . "/", "/display/" . $guid, $contact["url"] . "/");
+		if ($platform == 'friendica') {
+			return str_replace('/profile/' . $contact['nick'] . '/', '/display/' . $guid, $contact['url'] . '/');
+		}
+
+		if ($platform == 'hubzilla') {
+			return $contact['baseurl'] . '/item/' . $guid;
 		}
 
-		if (self::isRedmatrix($contact["url"])) {
-			return $contact["url"] . "/?mid=" . $guid;
+		if ($platform == 'socialhome') {
+			// This doesn't really seem to work on Socialhome
+			return $contact['baseurl'] . '/content/' . $guid;
```

Comment:
Is there something that would work better?

Reply:
False alarm. This had been a bug in my browser.